src/Components/SharedMain/SharedMain.tsx

```diff
@@ -1,14 +1,29 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import Cards from "../Cards/Cards";
 import styles from "./SharedMain.module.css";
-import { useSharedData } from "../../api/parseData";
 import FolderInfo from "../FolderInfo/FolderInfo";
 import Image from "next/image";
+import { SharedMainProps } from "@/types/interface";
+import { useSharedData } from "@/api/parseData";
+import { useFetch } from "@/hooks/useFetch";
+import { BASE_URL } from "@/constants/baseURL";
+
+const SharedMain: React.FC<SharedMainProps> = ({ folderId }) => {
+    const [userId, setUserId] = useState<number | null>(null);
+    const folderData = useFetch(`${BASE_URL}folders/${folderId}`);
+
+    useEffect(() => {
+        if (folderData && folderData.data && folderData.data.length > 0) {
```

Comment:
옵셔널체이닝(`?`) 사용해서 개선할 수 있겠네요 
```suggestion
        if (folderData?.data?.length > 0) {
```